jebenaclient/jebenaclient.py

```diff
@@ -564,6 +595,25 @@ def read_from_stdin(user_prompt=None):
     return "".join(reads)
 
 
+def __get_logger():
+    """Return a python logger for emitting logs."""
+    global __LOGGER
+    if not __LOGGER:
+        __LOGGER = logging.getLogger("jebenaclient")
+        # We define this "__get_logger" function in order to avoid triggering LOGGER stuff
+        # at import of jebenaclient. This allows us to run in py2 environments that have
+        # their own logging config to setup, which could run after this module is loaded.
+        # However, if the py2 env does not set up loggers, we need to do so before calling
+        # the logger to avoid spurious errors about unconfigured loggers.
+        # The cleanest way to do this is to 1) check if we're in Python 2; and if so,
+        # 2) fetch the root logger, and 3) if it has no handlers, call basicConfig
+        if sys.version_info[0] == 2:
+            if not len(logging.getLogger().handlers):
+                logging.basicConfig()
```

Comment:
are we sure this is going to work? i think last time we tried basicConfig it overrode all my custom logger stuff. @jpotter

Reply:
`logging.getLogger().handlers` should be non-empty if you have any custom loggers installed. This code only gets triggered after a call to `run_query()` is triggered, so as long as you've configured your loggers before executing the client, it _should_ be safe.

In theory. In practice, there might be something I've missed here, though -- let me know how it works for you?